File: RedSheet.DbEntities/Models/TeamRole.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Microsoft.EntityFrameworkCore;
using Rx.Core.Data.Attributes;
using RedSheet.DbEntities.Enums;

namespace RedSheet.DbEntities.Models
{
	[Table("TeamRoles", Schema = "dbo")]
	[Serializable]
	public partial class TeamRole  
	{
 
	[MaxLength(1000)]
	public string RoleDescription { get; set; } 
 
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)] 
	[Key]
	public int TeamRoleId { get; set; } 
 
	[Required] 
	[MaxLength(200)]
	public string TeamRoleName { get; set; } 
	[InverseProperty("TeamRole")]
	public ICollection<OurTeamMember> OurTeamMembers { get; set; } 


	public TeamRole()
	{
	this.OurTeamMembers = new HashSet<OurTeamMember>();

    }

    }
}

File: redsheet-portal/src/app/database-models/team-role.ts
import { required, maxLength, range, nested } from '@rx/annotations';
import { OurTeamMember,  } from './'
export class TeamRole {
    constructor(teamRole?: TeamRole )  {
        let properties = [ "roleDescription", "teamRoleId", "teamRoleName", "ourTeamMembers",];
        for (let property of properties)
            if (teamRole && teamRole[property])
                this[property] = teamRole[property];
    }
 
    @maxLength(1000)
	roleDescription : string =   undefined;
 
	teamRoleId : number =   0 ;
 
    @required()
    @maxLength(200)
	teamRoleName : string =   undefined;
	@nested(OurTeamMember)
	ourTeamMembers: OurTeamMember[];



}
